developer-guides/cosmwasm-contracts/compile-a-contract.md

```diff
@@ -0,0 +1,39 @@
+---
+description: How to Compile CosmWasm Smart Contracts
+---
+
+# Compile a Contract
+
+[This guide is from the official CosmWasm/rust-optimizer repository](https://github.com/CosmWasm/rust-optimizer). This method of compiling the contract will optimize the final build so that it reduces gas consumption.
+
+Example contracts for this can be found at [https://github.com/CosmWasm/cw-examples](https://github.com/CosmWasm/cw-examples). This repository requires the "Multiple Contract Repository" section.
+
+## Single Contract Repository
+
+The easiest way is to simply use the [published docker image](https://hub.docker.com/r/cosmwasm/rust-optimizer). You must set the local path to the smart contract you wish to compile and it will produce an `artifacts` directory with `<crate_name>.wasm` and `contracts.txt` containing the hashes. This is just one file.
+
```

Comment:
Section before here on an example of a project. Naming convention. "How to create your own To-Do-List" 

Something like ```cargo generate --git https://github.com/CosmWasm/cw-template.git --name PROJECT_NAME``` if i remember from my old times messing with CW. Maybe a whole new section on Create your own project using CW?

Reply:
yea this is less about creating and more so how to compile and deploy. I will link to book.cosmwasm.com in the compile stage. This is less about full help & more so just information is that is valuable